### DISCUSSION TITLE
How to sort fields with cursor based pagination ?

### DISCUSSION THREAD

**Discussion by NicoSan20** [sequelize/sequelize#14616] [Help & Questions] [answered]
Hello everyone ! I don't know if this is the best place for this question, but I'll try anyway because I'm completely lost :( Let me explain.

Currently, I'm trying to implement a cursor based pagination with sequelize (infinite scroll) and for that, here is what I did :
```
//...
const whereClause = {};

if (after) {
    whereClause['createdAt'] = { [Sequelize.Op.lt]: new Date(after) }; // <= after is my cursor
}

const products = await models.Product.findAll({
    order: [['createdAt', 'DESC']],
    limit: first + 1,
    where: whereClause,
});
//...
```
So far so good, as long as I sort by "createdAt" and my cursor is based on date.

Where it gets complicated is when I want to sort the result by "price" `order: [['price', 'DESC']]` and not by "createdAt". 

So my question is, can I do it with sequelize ? How can I sort by price ? How do I position my cursor ? (the purpose is not to load all the products, because is quite large)

Honestly, I don't know how to do it. Any help would be very appreciated, thank you in advance.

**Comment by ephys** [comment, marked as answer]
Stateless cursor based pagination is pretty complicated, if you want here is an outline of how to implement it: https://github.com/ephys/sequelize-cursor-pagination/blob/main/src/sequelize-find-by-cursor.ts#L287-L305

I'd like to add `Model.findByCursor` some day (don't quote me on this). In the meantime there are libraries that do it:

- https://github.com/ephys/sequelize-cursor-pagination (disclaimer: mine)
- https://www.npmjs.com/package/sequelize-cursor-pagination
